inference-engine/src/transformations/src/transformations/common_optimizations/common_optimizations.cpp

```diff
@@ -89,6 +90,11 @@ bool ngraph::pass::CommonOptimizations::run_on_function(std::shared_ptr<ngraph::
     // This pass must be called first in pipeline
     manager.register_pass<ngraph::pass::InitNodeInfo>();
     manager.register_pass<ngraph::pass::SimplifyShapeOfSubGraph>();
+
+    // This transformation must be called before constant folding as it folds
+    // ShapeOf node, which is required to be checked in dropout subgraph
+    manager.register_pass<ngraph::pass::DropoutWithRandomUniformReplacer>();
```

Comment:
It should not be moved to moc_transformatins because the plugin should decide when to enable this transformation and when not. The idea is that plugins which support RandomUniform operation should keep the graph as is and others which do not support may use this transformation to replace the sub-gpaph.

Reply:
Okay, I got it.
I actually don't like the idea that we start depending on CF here and we put transformation in the beginning of pipeline. And since we have this `DisableRandomUniformConstantFolding` pass the RundomUniform will never be folded so after CF in case with static shapes we will get Const->RU and ShapeOf->RU in case with dynamic shapes, so in DropoutWithRandomUniformReplacer pass pattern we can specify ShapeOf and Constant as inputs for RU (or we can put pattern::any_input() if RU first input doesn't make sense) and move this pass to the CommonDecompositions group. @popovaan @lazarevevgeny  what do you think?

Reply:
We discussed this transformation with @rkazants, ShapeOf can be removed from pattern. So in this case this transformation becomes independent from Constant Folding and can be moved to CommonDecompositions.

Reply:
Removed unnecessary nodes from pattern and moved transformation to CommonDecompositions. @GlebKazantaev please take a look.